core/templates/services/playthrough.service.ts

```diff
@@ -171,16 +176,21 @@ class MultipleIncorrectAnswersTracker {
   providedIn: 'root'
 })
 export class PlaythroughService {
-  private explorationId: string = null;
-  private explorationVersion: number = null;
-  private learnerIsInSamplePopulation: boolean = null;
-
-  private eqTracker: EarlyQuitTracker = null;
-  private cstTracker: CyclicStateTransitionsTracker = null;
-  private misTracker: MultipleIncorrectAnswersTracker = null;
-  private recordedLearnerActions: LearnerAction[] = null;
-  private playthroughStopwatch: Stopwatch = null;
-  private playthroughDurationInSecs: number = null;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  private explorationId!: string;
+  private explorationVersion!: number;
+  private learnerIsInSamplePopulation: boolean = false;
+
+  // Below properties are initialized with null values, and are set to non-null
+  // values only when the record action of corresponding exploration is starts.
```

Comment:
```suggestion
  // Below properties are initialized with null values and are set to non-null
  // values only when the record action of the corresponding exploration starts.
```
Can you please explain what do you mean by "record action" here?

Reply:
Done, is it clear now?